### PR TITLE
feat: add waive flag and payment functions

### PR DESCRIPTION
**Pull request by BLasan**
## Description
Payment and waive flag actions are added to the upcoming charges on general tab.

## Related issues and discussion
#691 

## Screenshots, if any
screenShot3

## Checklist
Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] If you have multiple commits please combine them into one commit by squashing them.

- [x] Read and understood the contribution guidelines at `web-app/.github/CONTRIBUTING.md`.

### CODE REVIEW

File: src/app/clients/clients-view/general-tab/general-tab.component.html [hunk 1]
```diff
@@ -54,10 +54,10 @@ <h3>Upcoming Charges</h3>
     <ng-container matColumnDef="Actions">
       <th mat-header-cell *matHeaderCellDef> Actions </th>
       <td mat-cell *matCellDef="let element">
-        <button class="account-action-button" mat-raised-button color="primary">
+        <button class="account-action-button" mat-raised-button color="primary" [routerLink]="['../','paycharge',element.id]">
```
Comment (karantakalkar) [inline src/app/clients/clients-view/general-tab/general-tab.component.html:57]: You should implement pay charge form as a part of this issue.
Screenshot from 2020-05-10 23-49-29

File: src/app/clients/clients.service.ts [hunk 2]
```diff
@@ -67,6 +67,13 @@ export class ClientsService {
     return this.http.get(`/clients/${clientId}/charges`, { params: httpParams });
   }
 
+  /**
+   * @param chargeData Charge Data to be waived.
+   */
+  waiveClientCharge(chargeData: any) {
+    return this.http.post(`/clients/${chargeData.clientId}/charges/${chargeData.resourceType}?command=waive`, chargeData);
```
Comment (karantakalkar) [inline src/app/clients/clients.service.ts:74]: use http params

File: src/app/clients/clients-view/general-tab/general-tab.component.ts [hunk 3]
```diff
@@ -57,4 +58,23 @@ export class GeneralTabComponent implements OnInit {
     this.showClosedShareAccounts = !this.showClosedShareAccounts;
   }
 
+  /**
+   * Waive Charge.
+   */
+  waiveCharge(chargeId: string, clientId: string) {
+    const waiveChargeObj = { clientId: clientId.toString(), resourceType: chargeId};
```
Comment (karantakalkar) [inline src/app/clients/clients-view/general-tab/general-tab.component.ts:65]: rename waiveChargeObj as 'charge' simply.

File: src/app/clients/clients-view/general-tab/general-tab.component.ts [hunk 4]
```diff
@@ -57,4 +58,23 @@ export class GeneralTabComponent implements OnInit {
...
+    this.clientService.waiveClientCharge(waiveChargeObj).subscribe(() => {
+      this.getChargeData(clientId);
+    });
+  }
+
+  /**
+   * Get Charge Data.
```
Comment (karantakalkar) [inline src/app/clients/clients-view/general-tab/general-tab.component.ts:72]: jsdoc param missing

File: src/app/clients/clients-view/general-tab/general-tab.component.ts [hunk 5]
```diff
@@ -57,4 +58,23 @@ export class GeneralTabComponent implements OnInit {
     this.showClosedShareAccounts = !this.showClosedShareAccounts;
   }
 
+  /**
+   * Waive Charge.
```
Comment (karantakalkar) [inline src/app/clients/clients-view/general-tab/general-tab.component.ts:62]: jsdoc params missing

**Comment by karantakalkar** [comment]
@BLasan lets get as many p1 PRs approved today, most of them are almost ready.

**Comment by BLasan** [comment]
> @BLasan lets get as many p1 PRs approved today, most of them are almost ready.

Sure. I'll refactor all of 'em

**Comment by karantakalkar** [comment]
@BLasan lets finalize this one

**Comment by BLasan** [comment]
> @BLasan lets finalize this one

Sure, I'm working on this

**Comment by BLasan** [comment]
@karantakalkar  Is this PR ok?

**Comment by karantakalkar** [comment]
we need to connect this once payment form is merged or vice versa

**Comment by BLasan** [comment]
> Thanks!

Welcome :)

**Comment by karantakalkar** [comment]
@muskankhedia review and merge, make sure to remove extea PR mention in commit message.

**Comment by muskankhedia** [comment]
A change in the route of the `paycharge` can be required based on the finalization of the UI in the PR #690. Merging this PR for now as that can be modified later in the PR #690 based on the decision and make the module functional.

**Pull request merged**